### ISSUE TITLE
Including notebooks with examples

### ISSUE THREAD

**Issue by jakirkham** [rapidsai/ucx-py#244]
After chatting with @randerzander and @VibhuJawa, we thought it might be nice to include some examples of how to use ucx-py. In particular we were thinking of usage with cuDF. Though other examples may make sense as well. Our thought was it would be good to create a `notebooks` directory here with a few examples. Does this seem reasonable to others? If so, what sorts of examples would we like to add here.

cc @quasiben @mrocklin

**Comment by mrocklin** [comment]
What kind of examples are we looking for?  If you're in a notebook using something like LocalCUDACluster then you just add a `protocol="ucx"` keyword

```python
from dask_cuda import LocalCUDACluster

cluster = LocalCUDACluster(protocol="ucx")
```

The challenge, I think, is setting up all of the UCX environment variables correctly (see #221 and https://github.com/openucx/ucx/issues/4273 and https://github.com/dask/distributed/pull/3135)

**Comment by jakirkham** [comment]
One example that was floated was doing a join with dask-cudf and UCX.

**Comment by mrocklin** [comment]
I think that such a notebook should exist.  It seems to be more about dask-cudf than UCX though.  Only one line in that notebook would include "ucx".  I think it makes sense for that to be in rapids notebooks instead.

**Comment by jakirkham** [comment]
I think the general consensus was we want to limit the number of places people need to look. So that would seem to meet that requirement. WDYT @randerzander? Would it be reasonable to stick this in [RAPIDS notebooks]( https://github.com/rapidsai/notebooks )?

**Comment by mrocklin** [comment]
I think also once ucx becomes easier to install and use (which is rapidly
becoming the case) we might also just make it on by default for all of the
dask-cuda options.

**Comment by randerzander** [comment]
Agreed that end-state a full-blown notebook is fine living in rapidsai/notebooks, and that existing notebooks should update their `dask_cuda` cluster instantiation to use UCX.

What I'm actually looking for is a sanity check example that confirms for me I have ucx installed and setup correctly such that when I run a dask join (or other highly communication impacted operation), I see it proceed faster than I did without ucx.

In its present state, the README in this repo explains how to install from conda, or from source, but the user is left without a clear way to test everything's working (whether over plain TCP, nvlink, or IB).

As a starting point just adding something like this at the bottom of the README would give a user some idea of how they can use UCX:
```
from dask_cuda import LocalCUDACluster
import dask_cudf

cluster = LocalCUDACluster(protocol="ucx")

# generate data in ddf1, ddf2

ddf = ddf1.merge(ddf2, on=['id'])
# run the distributed join operation
# workers communicate over nvlink and/or infiniband if available
%timeit ddf.compute()
```

**Comment by jakirkham** [comment]
Given best practices may change a bit here while we try and then settle upon the ideal solution, I'd still be tempted to include the notebook in ucx-py (at least near term). Once it is clear things work reliably, it would make sense to upstream this notebook to cuDF. WDYT @mrocklin?

**Comment by mrocklin** [comment]
I think that long term such a notebook shouldn't be here.  I'm happy to defer to your judgement on what is best and pragmatic short term.
